backends/blocked/ceed-blocked-operator.c

```diff
@@ -60,16 +60,25 @@ static int CeedOperatorSetupFields_Blocked(CeedQFunction qf,
       ierr = CeedElemRestrictionGetCeed(r, &ceed); CeedChk(ierr);
       CeedInt nelem, elemsize, nnodes;
       CeedInterlaceMode imode;
-      ierr = CeedElemRestrictionGetIMode(r, &imode); CeedChk(ierr);
       ierr = CeedElemRestrictionGetNumElements(r, &nelem); CeedChk(ierr);
       ierr = CeedElemRestrictionGetElementSize(r, &elemsize); CeedChk(ierr);
       ierr = CeedElemRestrictionGetNumNodes(r, &nnodes); CeedChk(ierr);
       ierr = CeedElemRestrictionGetNumComponents(r, &ncomp); CeedChk(ierr);
-      ierr = CeedElemRestrictionCreateBlocked(ceed, imode, nelem, elemsize,
-                                              blksize, nnodes, ncomp,
-                                              CEED_MEM_HOST, CEED_COPY_VALUES,
-                                              data->indices, &blkrestr[i+starte]);
-      CeedChk(ierr);
+      if (data->indices) {
+        ierr = CeedElemRestrictionGetIMode(r, &imode); CeedChk(ierr);
+        ierr = CeedElemRestrictionCreateBlocked(ceed, imode, nelem, elemsize,
+                                                blksize, nnodes, ncomp,
+                                                CEED_MEM_HOST, CEED_COPY_VALUES,
+                                                data->indices,
+                                                &blkrestr[i+starte]);
+        CeedChk(ierr);
+      } else {
+        CeedInt strides[3];
+        ierr = CeedElemRestrictionGetStrides(r, &strides); CeedChk(ierr);
+        ierr = CeedElemRestrictionCreateBlockedStrided(ceed, nelem, elemsize,
+               blksize, nnodes, ncomp, strides, &blkrestr[i+starte]);
+        CeedChk(ierr);
+      }
```

Comment:
For our blocked backends, I do not think we can us`CEED_STRIDES_BACKEND` to improve the amount of data movement at the moment. We may be able to reduce data movement for the serial backends (#250), but I don't think we would see a big impact unless we are recomputing qdata (or other inputs that use a strided restriction) frequently. For all CPU backends, we maintain a copy of the restricted input vector and use that appropriately permuted copy until the input changes.